### DISCUSSION TITLE
Dynamic Routes, force revalidate if 404

### DISCUSSION THREAD

**Discussion by FerrariAndrea** [vercel/next.js#59156] [Help]
### Summary

# Scenario introduction:
1. Next.js 14.0.3 (app)
2. Using a Dynamic Routes: /test/[slug]
3. Using import { notFound } from 'next/navigation', if the page doesn't exist (no data to show or other issue handled with 404)
4. The page has a long revalidate (8h): export const revalidate = 28800
5. The "fetch" that I'm using to retrieve data for building the page from the DB, has revalidated too: {next :{ revalidate: 14400}}
6. I'm also using "generateStaticParams" to generate at build time some of my pages under "/test/[slug]"

# Use case:

1. I visit a page that returns 404, calling notFound (). Like: /test/a  (where "a" is our [slug]), the page return 404 for example because the DB query returns no data, the data of the page are still not available
2. Now I know that the page now has data, so I return to the page "/test/a", but I still receive 404
3. if I come back to the page after 8h, the page works properly

# Looking for:

I'm looking for a solution that can help me, to ignore the revalidate if the page is under 404, so the page will build soon as possible when it can.

### Additional information

I tried several different configurations, and the best one was using this: [revalidatePath](https://nextjs.org/docs/app/api-reference/functions/revalidatePath)

but when the page can build I get a 500 with the error:
```
Page changed from static to dynamic at runtime /test/a, reason: revalidateTag _N_T_/test/a/page
see more here https://nextjs.org/docs/messages/app-static-to-dynamic-error
```

**Comment by fydellys** [comment]
I have the same problem. No solution so far.
